miprometheus/problems/video_to_class/seq_mnist_to_class/permuted_sequential_row_mnist.py

```diff
@@ -132,13 +132,13 @@ def __getitem__(self, index):
         label = self.labels[target.data]
 
         # create mask
-        mask = torch.zeros(self.num_rows).type(self.app_state.IntTensor)
-        mask[-1] = 1
+        mask = torch.zeros((self.num_rows,1)).type(self.app_state.IntTensor)
+        mask[-1,0] = 1
 
         data_dict = DataDict({key: None for key in self.data_definitions.keys()})
-        data_dict['images'] = img
+        data_dict['images'] = img.view(28,1,1,28)
         data_dict['mask'] = mask
-        data_dict['targets'] = target
+        data_dict['targets'] = target*torch.ones((28,1),dtype=torch.long)
```

Comment:
Ok, so the trick is that you are using here torch.long, whereas mask is of type: self.app_state.IntTensor

Generally self.app_state.IntTensor is a remainer of our previous solution, where app_state stored cpu/gpu types, depending on the setting. This appeared to be problematic when we started to use DataLoaders with multiple workers - we had issues with different loader workers tried to load different images to GPU directly.

So instead we decided that: getitem/collate return items/batches that remain in CPU memory, then the mip-worker transports them to GPU (DataDict has a function that does that automatically).

Solution: please replace all references to self.app_state.TYPE with torch.TYPE.

@vmarois Can you please confirm that?

Reply:
Yes, I confirm this indeed. This is the line which converts the cpu tensor types to CUDA ones:

https://github.com/IBM/mi-prometheus/blob/master/miprometheus/workers/worker.py#L474

So batch generation stays on cpu for now, but we might want to change that in the future if we use [PyTorch's DataParallel](https://pytorch.org/docs/stable/nn.html?highlight=dataparallel#torch.nn.DataParallel) object